[WebKit checkers] Recognize NS_RETURNS_RETAINED and CF_RETURNS_RETAINED.

rniwa opened the pull request:
This PR adds the support for treating a function return value to be safe if the function is annotated with NS_RETURNS_RETAINED or CF_RETURNS_RETAINED.

rniwa committed:
[WebKit checkers] Recognize NS_RETURNS_RETAINED and CF_RETURNS_RETAINED.

This PR adds the support for treating a function return value to be safe
if the function is annotated with NS_RETURNS_RETAINED or CF_RETURNS_RETAINED.

t-rasmud commented on clang/test/Analysis/Checkers/WebKit/unretained-call-args.mm:
```diff
@@ -438,6 +438,20 @@ void use_const_local() {
+}
+
+} // namespace ns_retained_return_value
+
```
I'm fairly certain this works, but can we have a test case with inheritance (sub class calls `provideNS` defined in super class) for the sake of completion?

rniwa replied:
You mean `provideNS` returns a subclass of a class `consumeNS` takes?

t-rasmud reviewed (approved):
I left a small nit in the test case. Otherwise, LGTM!

rniwa committed:
Add a test case per review comment.
Define a function which returns a retained value in a base class and call that in a derived class
